Fix exceptions from pre-commit

hillaryj opened the pull request:
Fixes all the pre-commit error messages - not sure why it didn't run before pre-commit, but it's there now!

## 🗣 Description

Lots of small changes, mostly EOF and for shellcheck.

## 💭 Motivation and Context

Fixes the automation that we rely on.

## 🧪 Testing

Tested that these files load correctly on my local terminal.

## 🚥 Types of Changes

<!--- What types of changes does your code introduce? -->
<!--- Put an `x` in all the boxes that apply: -->

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

<!--- Go over all the following points, and put an `x` in all the
boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask.
We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

hillaryj committed:
Fix exceptions from pre-commit

hillaryj committed:
Prevent shellcheck on all zsh files

jsf9k commented on bash/.bashrc:
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+# shellcheck disable=1090
```
Is it possible to disable just at the line where it is needed, instead of globally?

hillaryj replied:
Refined in https://github.com/cisagov/.dotfiles/pull/2/commits/e4d2a6d4a48c1b66d6ff25dfc8baf93d808fe876

jsf9k commented on deploy.sh:
```diff
@@ -6,9 +6,11 @@ set -o pipefail
 
 echo "Finding dotfiles to deploy via stow..."
 
-for d in $(find . -maxdepth 1 -mindepth 1 -type d -not -name ".git" -exec basename {} \;)
```
Why doesn't pre-commit like this `for` loop?  I think this is a clearer way of doing it than using a heredoc.

hillaryj replied:
Shellcheck recommends not using `for` loops due to fragility - this is one of their recommended POSIX-compliant patterns from https://github.com/koalaman/shellcheck/wiki/SC2044

I would also be happy to revert and just `disable=SC2044` if y'all prefer.

jsf9k replied:
I guess they have a point, even if the result is a little ugly.  In your case, though, I _think_ you can completely get rid of the `for` loop:
```
find . -maxdepth 1 -mindepth 1 -type d -not -name ".git" -exec stow -v -R -t ~ $(basename {}) \;
```

Does that work?

hillaryj replied:
I got a note that `This will expand once before find runs, not per file found`

jsf9k replied:
I think that message is a red herring.  Enclosing `$(basename {})` in quotes seems to do the trick.  See commit c2cf909, and let me know if you're OK with that.

hillaryj replied:
Looks great! I've tested it as well and it works well. Thanks!

jsf9k commented on zsh/.zshrc.d/prompt:
```diff
@@ -1,5 +1,6 @@
 #!/bin/zsh
 # shellcheck shell=bash
+# shellcheck disable=SC2034
```
Don't disable this globally.  Just disable it on the lines where it is needed.

hillaryj replied:
I'll remove it entirely now that I've excluded the `zsh` files.

hillaryj replied:
Fixed in https://github.com/cisagov/.dotfiles/pull/2/commits/e4d2a6d4a48c1b66d6ff25dfc8baf93d808fe876

jsf9k reviewed (changes requested):
I have a few issues/questions.  Please see my comments.

hillaryj committed:
Refine shellcheck directives

hillaryj committed:
Fix shellcheck comment

hillaryj committed:
Remove comment after shellcheck directive

jsf9k committed:
Get rid of shellcheck complaint

shellcheck doesn't like for find to feed a for loop, but in this case
that isn't necessary.

dav3r commented on bash/.bashrc.d/prompt:
```diff
@@ -12,23 +14,23 @@ RESET=$(tput sgr0)
 
 LINECHAR="─"
-PREFIX="❱" # U+2771
+PREFIX_CHAR="❱" # U+2771
```
Style thang here, but `LINECHAR` should probably now be renamed to `LINE_CHAR` to match `PREFIX_CHAR` and the rest of the variables in here.

hillaryj replied:
Resolved in https://github.com/cisagov/.dotfiles/pull/2/commits/069150a9663c4302f2804952169be9bc2e2c2fe6

dav3r reviewed (approved):
LGTM!  🏎️ 
Please consider my one minor comment.

hillaryj committed:
Update variable for name consistency
